app/src/main/java/com/mikepenz/fastadapter/app/SortActivity.java

```diff
@@ -0,0 +1,211 @@
+
+        //create our FastAdapter which will manage everything
+        fastItemAdapter = new FastItemAdapter<>();
+
+        //configure our fastAdapter
+        fastItemAdapter.withOnClickListener(new FastAdapter.OnClickListener<SampleItem>() {
+            @Override
+            public boolean onClick(View v, IAdapter<SampleItem> adapter,
+                                   SampleItem item, int position) {
+                Toast.makeText(v.getContext(), (item).name.getText(v.getContext()),
+                        Toast.LENGTH_LONG).show();
+                return false;
+            }
+        });
+
+        //get our recyclerView and do basic setup
+        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.rv);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setAdapter(fastItemAdapter);
+
+        if (savedInstanceState != null) {
+            //noinspection unchecked
+            fastItemAdapter.getItemAdapter().withComparator((Comparator<SampleItem>)
+                    savedInstanceState.getSerializable("comparator"));
+        }
+
+        fastItemAdapter.setNewList(generateUnsortedList());
+
+        //restore selections (this has to be done after the items were added
+        fastItemAdapter.withSavedInstanceState(savedInstanceState);
+
+        //set the back arrow in the toolbar
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setHomeButtonEnabled(false);
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        //add the values which need to be saved from the adapter to the bundle
+        outState = fastItemAdapter.saveInstanceState(outState);
+        //We need to persist our comparator between orientation changes
+        outState.putSerializable("comparator", (Serializable)
+                fastItemAdapter.getItemAdapter().mComparator);
+        super.onSaveInstanceState(outState);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        //handle the click on the back arrow click
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                Toast.makeText(getApplicationContext(), "selections = " +
+                        fastItemAdapter.getSelections(), Toast.LENGTH_LONG).show();
+                onBackPressed();
+                return true;
+
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    /**
+     * Sorts the list ascending
+     */
+    @OnClick(R.id.button_ascending)
+    public void sortAscending() {
+        //Set the new comparator to the list
+        fastItemAdapter.getItemAdapter().withComparator(new AlphabetComparatorAscending());
+        //Reset the list to trigger the sorting
+        fastItemAdapter.setNewList(generateUnsortedList());
```

Comment:
Do we really need to reset the list here? Isn't there a method like `resort()` or something?
`notifyDataSetChanged()` did not help.